File: src/core/js/pipe.js
const _core_pipe = (b, ...a) => {
  let result = b;
  for (let i = 0; i < a.length; i++) {
    result = a[i](result);
  }
  return result;
};

const core_pipe = (...a) => {
  if ('function' === typeof a[0]) {
    return (b) => _core_pipe(b, ...a);
  } else {
    return _core_pipe(...a);
  }
};

File: src/core/py/pipe.py
def _core_pipe(a, *fn):
  result = a
  for i in range(0, len(fn)):
    result = fn[i](result)
  
  return result

def core_pipe(*p):
  if isinstance(p[0], types.LambdaType):
    return lambda a : _core_pipe(a, *p)
  else:
    return _core_pipe(*p)
